src/App/Fossa/ProjectInference.hs

```diff
@@ -24,9 +26,16 @@ import Text.GitConfig.Parser (Section (..), parseConfig)
 import Text.Megaparsec (errorBundlePretty)
 import Effect.ReadFS
 
+mergeOverride :: OverrideProject -> InferredProject -> ProjectRevision
+mergeOverride OverrideProject {..} InferredProject {..} = ProjectRevision name revision branch
+  where
+    name = fromMaybe inferredName overrideName
+    revision = fromMaybe inferredRevision overrideRevision
+    branch = fromMaybe inferredBranch overrideBranch
```

Comment:
@cnr There are cases where we have the branch, but never actually use it.  Does this break the laziness we had by previously not having access to it at all?